### PR TITLE
Ensure a single transaction per time

### PR DESCRIPTION
**Pull request by magestrio**
## Type

- [ ] Feature
- [X] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/compile-service/src/index.ts [hunk 1]
```diff
@@ -33,50 +33,41 @@ async function findTransactionWithRetry(
...
 
-  if (!serviceDb.isConnected()) {
-    await serviceDb.connect();
-  }
-
-  if (!proofDb.isConnected()) {
-    await proofDb.connect();
-  }
+  if (!serviceDb.isConnected()) await serviceDb.connect();
+  if (!proofDb.isConnected()) await proofDb.connect();
```
Comment (SangTran-127) [inline packages/compile-service/src/index.ts]: If block

File: packages/compile-service/src/index.ts [hunk 2]
```diff
@@ -33,50 +33,41 @@ async function findTransactionWithRetry(
...
   const modelTransaction = ModelDbTransaction.getInstance();
   const modelDbSettings = ModelDbSetting.getInstance();
 
   while (true) {
-    const request = await redisQueue.dequeue();
-    if (request) {
+    try {
+      const request = await redisQueue.dequeue();
+      if (!request) {
+        await setTimeout(1000); // Prevent busy looping when the queue is empty
```
Comment (SangTran-127) [inline packages/compile-service/src/index.ts:76]: I think you should use global variable for timeout

File: packages/compile-service/src/index.ts (hunk 1 above)
Comment (magestrio) [reply packages/compile-service/src/index.ts]: done

File: packages/compile-service/src/index.ts (hunk 2 above)
Comment (magestrio) [reply packages/compile-service/src/index.ts]: done

**Review by SangTran-127 — APPROVED** [review]

**Review by SangTran-127 — APPROVED** [review]

**Pull request merged**